<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewConnectionDialog</class>
 <widget class="QDialog" name="NewConnectionDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>696</width>
    <height>633</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create a New Connection</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QGroupBox" name="connection_box">
     <property name="title">
      <string>Connection Information</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="2">
       <widget class="QLineEdit" name="name_edit">
        <property name="toolTip">
         <string>Name of the STAC API connection.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>Planetary Computer</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_url">
        <property name="text">
         <string>URL</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLineEdit" name="url_edit">
        <property name="toolTip">
         <string>URL of the connection.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>http://example.com/stac-endpoint/</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" rowspan="3" colspan="3">
       <widget class="QgsCollapsibleGroupBox" name="collections_group">
        <property name="title">
         <string>Advanced</string>
        </property>
        <property name="collapsed">
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="sas_subscription_key_la">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Subscription key that can be used when signing SAS based connection items.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>SAS subscription key</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="sas_subscription_key">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Subscription key that can be used when signing SAS based connection items.</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="result_collections_la">
           <property name="text">
            <string>API Conformance classes</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QTreeView" name="conformances_tree">
             <property name="toolTip">
              <string/>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::MultiSelection</enum>
             </property>
             <property name="indentation">
              <number>5</number>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
             <property name="allColumnsShowFocus">
              <bool>true</bool>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="get_conformances_btn">
             <property name="toolTip">
              <string>Get this STAC API conformances, as specified in the API root page. </string>
             </property>
             <property name="text">
              <string>Get conformance classes</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Options</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="support">
             <property name="toolTip">
              <string>Select other capabilities that the API supports.</string>
             </property>
             <property name="text">
              <string>API Capabilities</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="capabilities"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Page size</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QgsSpinBox" name="page_size">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_name">
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item row="9" column="2">
       <widget class="QDialogButtonBox" name="buttonBox">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="standardButtons">
         <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="2">
       <widget class="QgsAuthConfigSelect" name="auth_config">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="7" column="1" colspan="2">
       <widget class="QPushButton" name="test_btn">
        <property name="text">
         <string>Test connection</string>
        </property>
       </widget>
      </item>
      <item row="8" column="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsAuthConfigSelect</class>
   <extends>QWidget</extends>
   <header>qgsauthconfigselect.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>NewConnectionDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>550</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>NewConnectionDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>550</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
